packages/ember-extension-support/lib/models/inspected_node.js

```diff
@@ -0,0 +1,234 @@
+import Component from "ember-views/views/component";
+import { A as emberA } from "ember-runtime/system/native_array";
+import { get } from "ember-metal/property_get";
+
+/**
+ * Wraps a render node to provide debugging API for external services
+ * such as the Ember Inspector.
+ *
+ * Ideally public methods on this class should completely hide
+ * the internal implementations of render nodes.
+ *
+ * @class InspectedNode
+ * @param {Object} renderNode The Htmlbars render node
+ * @param {Object} parent The parent InspectedNode instance
+ */
+function InspectedNode(renderNode, parent) {
+  this.renderNode = renderNode;
+  this.parent = parent;
+}
+
+InspectedNode.prototype =  {
+  /**
+   * Useful for subclassing.
+   *
+   * @property constructor
+   * @type {Function}
+   */
+  constructor: InspectedNode,
+
+  /**
+   * The render node we are representing and adding
+   * debugging functionality to.
+   *
+   * This property is set upon creation of the instance.
+   *
+   * @property renderNode
+   * @type {Object}
+   * @private
+   */
+  renderNode: null,
```

Comment:
The downside of making this inaccessible is that I can't match render nodes that I got from perf instrumentation to the render nodes in this view tree.  (I use this to add render times next to the views in the view tree). Unless if we maybe provide the render node's `guid` here and in perf instrumentation?